### ISSUE TITLE
Define format and behaviour for uploading tags with XCA

### ISSUE THREAD

**Issue by mwinokan** [m2ms/fragalysis-frontend#1196]
Ryan was uploading new data to the LHS for D68 and observed the following occurring with tags found in the metadata.csv:

* New structures had their tag uploaded correctly
* For re-uploads of existing structures, fragalysis seemed to pull historic tags from a previous upload, but also added additional tags with no known origin

Our suggestion would be that in future:

* The tag/site column can be left blank in the uploaded metadata.csv, in which case the tags as existing in Fragalysis should be used
* In case tags are uploaded in the CSV, these should overwrite whatever is in Fragalysis.
* It should be possible to upload multiple tags in the CSV, semi-colon-separated or otherwise.
* When downloading structures, the metadata.csv should reflect exactly the tags as present in Fragalysis.

I'm not sure whether this is still a problem in Fragalysis 2.0, but just documenting the present issues.

**Comment by mwinokan** [comment]
@tdudgeon & @alanbchristie please confirm whether this needs fixing in Fragalysis 2.0 or if tags have been reworked with consistent behaviour already.

**Comment by phraenquex** [comment]
@kaliif, Warren just alerted me to this - it is in fact a really really critical feature that cannot be broken in V2. 

(We almost certainly needn't fix it in V1, because we shouldn't be uploading to V1 ever again.)

**Comment by kaliif** [comment]
@mwinokan, @phraenquex, tags are handled very differently in v2. There's no metadata.csv,  they are generated from the uploaded data directly. Correctly or not, that's something I cannot tell. Needs to be verified by someone who understands them.

There are two targets now in the staging stack, if somebody could have a look?

**Comment by phraenquex** [comment]
Thanks @kaliif.  One Very Important Bug in v1 is that existing tags are killed by new uploads.  Very Evil.

There does need to be a mechanism for uploading custom tag assignments too, alongside those auto-generated from the XCA categories.  (e.g. metadata extracted from SoakDB files.)  
* You get to define the mechanism by which they should be added to the XCA needs to be defined - csv probably easiest (else yaml?) - format should be well-defined.  
* You do NOT need to define the mechanism by which that CSV (or whatever) gets generated.

Do create a separate ticket if necessary.

**Comment by mwinokan** [comment]
@kaliif We will need to upload LHS data for D68EV3CPROA this week and it would be great if we can avoid having to manually re-tag the 200+ hits again. Is it feasible to fix the issues in v1 this week?

**Comment by mwinokan** [comment]
I just spoke to @phraenquex and we will **not** be making any changes to v1

**Comment by phraenquex** [comment]
This should fall under #1226.

This might be a marginal use-case for now, because there are no upstream tools for generating non-auto tags.

However, we do have to pin down the format and behaviour.  And that's in scope for this ticket.

**Comment by mwinokan** [comment]
@mwinokan Check if tag behaviour in #1311 will be sufficient

**Comment by mwinokan** [comment]
No longer needed after #1263
